securedrop/requirements/update_python_dependencies

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Usage: ./update-python-dependencies
+# Run periodically to keep Python application, test, and development
+# requirements up-to-date.
+
+set -eC
+
+WORKING_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+declare -r WORKING_DIR
+declare -r VENV='review_env'
+declare -ra REQUIREMENTS_FILE_PREFIXES=('securedrop' 'test' 'develop')
+
+# This script should not be run with an active virtualenv. Calling deactivate
+# does not work reliably, so instead we quit with an error message.
+if [[ -n "${VIRTUAL_ENV}" ]]; then
+  printf '\nERROR: Please deactivate your virtualenv then re-run this script.\n'
+  exit 1
+fi
+
+# If pip and virtualenv are not available, try to install them with apt-get. If
+# apt-get is not available, quit with an error message.
+if ! command -v pip virtualenv > /dev/null; then
+  if command -v sudo apt-get > /dev/null; then
+    sudo apt-get -qq install python-virtualenv python-pip
+  else
+    printf '\nERROR: Virtualenv and pip must be installed to run this script.\n'
+    exit 1
+  fi
+fi
+
+pushd "${WORKING_DIR}"
+# Create a temporary virtualenv for the SecureDrop Python packages in our
+# requirements directory and make sure it's cleaned up on exit.
+trap 'rm -rf "${VENV}" && popd' EXIT
+
+virtualenv -p python2 "${VENV}"
+# shellcheck disable=SC1090
+source "${VENV}"/bin/activate
+pip install -U pip-tools
```

Comment:
I feel like you misunderstood my review comment. I'm not saying we should do away with `update_python_dependencies`—I think it's great. I just think it might be kind of unnecessary (and a waste of time, over repeated runs) to create (then destroy) a new virtualenv *just* to install pip-tools every time this script is run.

`pip-tools` is already included in `develop-requirements.in`. Couldn't we just update the docs to say that `update_python_dependencies` has to be run from an environment where those requirements are installed (e.g. as we already do in https://docs.securedrop.org/en/latest/development/documentation_guidelines.html), and then simply the script a bit?

I just really don't see the point of creating a virtualenv, installing pip-tools, then destroying it again. Is there some use case I am missing?